server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -258,42 +254,38 @@ TopDocs mergeTopDocs(Collection<TopDocs> results, int topN, int from) {
             final TopFieldDocs[] shardTopDocs = results.toArray(new TopFieldDocs[numShards]);
 
             int sortLength = shardTopDocs[0].fields.length;
+            final String className = SortedNanosecondsNumericSortField.class.getName();
             for (int sortIdx = 0; sortIdx < sortLength; sortIdx++) {
                 final int idx = sortIdx;
 
                 boolean containsNanosecond = Arrays.stream(shardTopDocs)
                     .filter(d -> d.totalHits.value > 0)
-                    .map(d -> d.fields[idx].getClass().getName())
-                    .anyMatch(className -> SortedNanosecondsNumericSortField.class.getName().equals(className));
+                    .anyMatch(d -> className.equals(d.fields[idx].getClass().getName()));
```

Comment:
nit: you can do `d.fields[idx].getClass().equals(SortedNanosecondsNumericSortField.class)` ?

Reply:
fixed